lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -444,6 +447,267 @@ TfTokenVector getProxyShapePurposes(const Ufe::Path& path)
     return purposes;
 }
 
+namespace {
```

Comment:
A large part of the code below was moved from ufe/private/Utils.cpp to centralize all restriction code in the same place.